pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
so you're calling d.CheckImage which in turn is calling kube.InspectImage and kube.InspectImage is eating the dockerapi imageNotFoundError and instead returning a kube.imageNotFoundError.

it's the kube.imageNotFoundError that has "no such image" (lowercase) as its message output.

so as expected, you're getting a kube.imageNotFoundError.

but you're right, i thought i saw a method for checking the error type in kube, but i was wrong, it was something else, i don't see any way to confirm the error type other than, as you said, checking the message text, which is bad.